app/questionnaire/path_finder.py

```diff
@@ -34,31 +36,69 @@ def routing_path(
         """
         Visits all the blocks in a section and returns a path given a list of answers.
         """
-        blocks: List[Mapping] = []
-        routing_path_block_ids: List[str] = []
+        routing_path_block_ids: list[str] = []
         current_location = Location(section_id=section_id, list_item_id=list_item_id)
         section = self.schema.get_section(section_id)
         list_name = self.schema.get_repeating_list_for_section(
             current_location.section_id
         )
 
         if section:
+            when_rules_block_ids_dependencies = (
+                self._get_when_rules_block_ids_dependencies(section["id"])
+            )
+            blocks = self._get_not_skipped_blocks_in_section(
+                current_location,
+                routing_path_block_ids,
+                section,
+                when_rules_block_ids_dependencies,
+            )
+
+            if blocks:
+                routing_path_block_ids = self._build_routing_path_block_ids(
+                    blocks, current_location, when_rules_block_ids_dependencies
+                )
+
+        return RoutingPath(routing_path_block_ids, section_id, list_item_id, list_name)
+
+    def _get_when_rules_block_ids_dependencies(self, section_id: str) -> list[str]:
+        """NB: Does not support repeating sections"""
```

Comment:
I think a bit more would help. This makes me think repeating doesn't work at all, but it's to make it clear that, a repeating section can be a dependent but not depended upon.